jumpstarter/states.py

```diff
@@ -0,0 +1,152 @@
+from enum import Enum, auto
+
+import transitions
+from transitions.extensions.nesting import NestedState
+from transitions_anyio import HierarchicalAnyIOMachine
+
+NestedState.separator = "↦"
+
+
+class ActorStartingState(Enum):
+    dependencies_started = auto()
+    resources_acquired = auto()
+    tasks_started = auto()
+
+
+class ActorStoppingState(Enum):
+    tasks_stopped = auto()
+    resources_released = auto()
+    dependencies_stopped = auto()
+
+
+class ActorRestartingState(Enum):
+    starting = auto()
+    stopping = auto()
+
+
+class ActorState(Enum):
+    initializing = auto()
+    initialized = auto()
+    starting = ActorStartingState
+    started = auto()
+    stopping = ActorStoppingState
+    stopped = auto()
+    restarting = ActorRestartingState
+    crashed = auto()
+
+
+class ActorStateTriggers(str, Enum):
+    def _generate_next_value_(name, *args):
+        return name.lower()
+
+    initialize = auto()
+    start = auto()
+    stop = auto()
+    restart = auto()
+    report_error = auto()
+
+
+class ActorStateMachine(HierarchicalAnyIOMachine):
+    def __init__(self, actor_state=ActorState):
+        super().__init__(
+            states=actor_state,
+            initial=actor_state.initializing,
+            auto_transitions=False,
+            send_event=True,
+        )
+
+        # TODO: Is there a way to do this with enums?
```

Comment:
Yes, there is.
You need to nest enums.
See states.py.

Reply:
Nesting is not the issue. Providing additional params like `initial`. The current working solution I have relies upon that.